### DISCUSSION TITLE
DRY isAllowedMethod util

### DISCUSSION THREAD

**Discussion by Vandivier** [vercel/next.js#25599] [Ideas]
### Describe the feature you'd like to request

I'm happy to PR this and have code ready, just want to document...

isAllowedMethod is copied and pasted in all three subframeworks of Commerce (BigCommerce, Shopify, Swell)

The requested feature is useful when, for instance, adding a new subframework that also wants to use the isAllowedMethod. Currently, both options are awkward:
1. have the same file N times
2. import from an arbitrary sibling instead of a higher-level framework

### Describe the solution you'd like

the simple request is to have a top-level /utils/is-allowed-method.ts which they all point to instead of having the duplicate code.

### Describe alternatives you've considered

1. leaving three files and the import from a top-level /utils/is-allowed-method.ts and having N duplicate files
2. moving into higher-level Commerce API; this isn't great either bc I would like my own framework which is independent from Commerce from an information architecture perspective; commerce and my own app API would both use top-level framework utils is a better organization

**Comment by Vandivier** [comment]
this isn't a nextJs-specific issue, but one for the commerce starter specifically. continue the convo here:
https://github.com/vercel/commerce/issues/224

**Comment by lfades** [comment]
Answered here: https://github.com/vercel/commerce/issues/224#issuecomment-853474221. Also this is unrelated to Next.js but related to Vercel Commerce so keep the conversation there or otherwise I might never find this.
